build_tools/buildkite/wait_for_pipeline_success.py

```diff
@@ -0,0 +1,225 @@
+          buildkite.BuildState.SKIPPED,
+          buildkite.BuildState.NOT_RUN,
+      ]:
+        output_str = f"Build finished in state '{state.name}'"
+        min_line_length = max(min_line_length, len(output_str))
+        print(output_str.ljust(min_line_length))
+        return state
+
+      output_str = (
+          f"Waiting for build {build_number} to complete. Waited {wait_time}"
+          f" seconds. Currently in state '{state.name}':"
+          f" {self.get_url_for_build(build_number)}")
+      min_line_length = max(min_line_length, len(output_str))
+      print(output_str.ljust(min_line_length), "\r", end="", flush=True)
+      # Yes, polling is unfortunately the best we can do here :-(
+      time.sleep(5)
+
+  def get_url_for_build(self, build_number):
+    return f"https://buildkite.com/{self._organization}/{self._pipeline}/builds/{build_number}"
+
+
+def parse_args():
+  parser = argparse.ArgumentParser(
+      description="Waits on the status of the last BuildKite build for a given"
+      " commit or creates such a build if none exists")
+  parser.add_argument(
+      "pipeline", help="The pipeline for which to create and wait for builds")
+  parser.add_argument(
+      "--rebuild",
+      help="Behavior for triggering a new build even if there is an existing"
+      " one. `force`: always rebuild without checking for existing build,"
+      " `failed`: rebuild on build finished in 'failed' state, `bad`: rebuild"
+      " on build finished in state other than 'passed'",
+      choices=["force", "failed", "bad"],
+  )
+  return parser.parse_args()
+
+
+def main(args):
+  bk = BuildkitePipelineManager.from_environ(args.pipeline)
+  build = bk.get_latest_build()
+  create_new_build = True
+
+  if build:
+    create_new_build = False
+    state = buildkite.BuildState(build["state"])
+    build_number = get_build_number(build)
+    url = bk.get_url_for_build(build_number)
+    print(f"Found previous build with state '{state.name}': {url}")
+
+    if args.rebuild == "force":
+      print(f"Received `--rebuild=force`, so creating a new build")
+      create_new_build = True
+    elif args.rebuild == "failed" and state == buildkite.BuildState.FAILED:
+      print(f"Previous build failed and received `--rebuild=failed`, so"
+            f" creating a new one.")
+      create_new_build = True
+    elif args.rebuild == "bad" and state in (
+        buildkite.BuildState.FAILED,
+        buildkite.BuildState.CANCELED,
+        buildkite.BuildState.SKIPPED,
+        buildkite.BuildState.NOT_RUN,
+    ):
+      print(f"Previous build completed with state '{state.name}' and received"
+            f" `--rebuild=bad`, so creating a new one.")
+      create_new_build = True
+  else:
+    print("Didn't find previous build for pipeline. Creating a new one.")
+
+  if create_new_build:
+    build = bk.create_build()
+  build_number = get_build_number(build)
+  state = bk.wait_for_build(build_number)
+
+  build_number = get_build_number(build)
```

Comment:
Do we need to get the build_number twice? (line 213)

Reply:
We surely do not :-) Thanks for catching.